examples/scrypt/pooled.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const Piscina = require('../../dist/src');
+const { resolve } = require('path');
+const crypto = require('crypto');
+const { promisify } = require('util');
+const randomFill = promisify(crypto.randomFill);
+const { performance, PerformanceObserver } = require('perf_hooks')
+
+const obs = new PerformanceObserver((entries) => {
+  console.log(entries.getEntries()[0].duration)
+});
+obs.observe({ entryTypes: ['measure']});
+
+const piscina = new Piscina({
+  filename: resolve(__dirname, 'scrypt.js'),
+  concurrentTasksPerWorker: 10
+});
+
+async function* generateInput() {
+  let max = 1000;
+  const data = Buffer.allocUnsafe(10);
+  while (max-- > 0) {
+    yield randomFill(data);
+  }
+}
+
+(async function() {
+  performance.mark('start');
+  const keylen = 64;
+  const salt = Buffer.allocUnsafe(16);
+  await randomFill(salt);
```

Comment:
Not sure how example-y we want this to be, but re-using a salt is not generally a good idea ;)

Reply:
Yeah, I pulled that out to isolate the number of Async tasks happening during the measuring but can definitely move that beck in.